tcpclient/connectionsManagement_test.go

```diff
@@ -13,41 +12,93 @@ import (
 // We really need to refactor this test. We should verify connections do become established,
 // rather than just waiting for a second and finish
 // We should also test "failing" connections, and ensure their status is reported properly
-func TestTcpConnect(t *testing.T) {
-	var numberConnections = 2
+func TestTcpConnectEstablished(t *testing.T) {
 	var host = "127.0.0.1"
 	var port = 55555
 
 	dispatcher := &tcpserver.Dispatcher{make(map[string]*tcpserver.Handler)}
 
 	run := func() {
+		t.Log("Starting TCP server...")
 		if err := dispatcher.ListenHandlers(port); err != nil {
-			t.Error("Could not start the TCP server", err)
+			t.Fatal("Could not start the TCP server", err)
 			return
 		}
-		t.Log("TCP server started")
 	}
 	go run()
```

Comment:
give "run()" a better name, or just do not store the function in a variable (call it directly in a goroutine)

Reply:
ack